packages/ui/src/Card.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC } from 'react'
+import { Icon, IconProps } from './Icon'
+import { DataTestProp } from '@hazelcast/helpers'
+import cn from 'classnames'
+
+import { IconButton, IconButtonProps } from './IconButton'
+
+import styleConsts from '../styles/constants/export.scss'
+import styles from './Card.module.scss'
+
+type CardHeaderProps =
+  | {
+      title: string
+      icon?: IconProps['icon']
+      iconButton?: never
+    }
+  | {
+      title: string
+      iconButton?: Omit<IconButtonProps, 'size' | 'kind'>
```

Comment:
Should we rename it to `iconButtonProps`?

Reply:
It's literally props of `IconButton` component. When I see `iconButton` prop I instantly think about `IconButton` itself.
For `icon` we are picking a single prop which is a component so it makes sense to call it `icon` and not `iconProps`.

Reply:
Gues that makes sense, I'll update it. Thanks ❤️